### DISCUSSION TITLE
Providing crash report

### DISCUSSION THREAD

**Discussion by p808u2** [ratgdo/homekit-ratgdo#177] [Ideas]
I noticed a crash report from the UI. I can't understand what happened myself, but I don't know how useful it is if I can't open an issue. Perhaps offer a way to automatically/manually send the crash report if they are valuable to the developer team?

Not opening an issue, but I've attached one log here for reference.

[ratgdo_hk_log_crash_May9.txt](https://github.com/ratgdo/homekit-ratgdo/files/15276929/ratgdo_hk_log_crash_May9.txt)

**Comment by dkerr64** [comment]
Thank you for the crash log.  Unfortunately I cannot tell much from it.  The crash was very early in the startup process (less than one minute in) and I can't see anything going on in the stack trace or the message log.  I suggest you clear the crash log and see how things go.  If it occurs again report back.

@jgstroud we need to add the firmware version number somewhere in the crash log... so we know what elf file to decode the stack against.  I think easiest done in the callback where I save the message log buffer.

**Reply by dkerr64** [reply]
@jgstroud FYI, I have added firmware version to crash log in next-release.

**Reply by jgstroud** [reply]
Cool.  I tried adding it and for whatever reason I just couldn't get it to work.  It was failing trying to move it from the PSTR or something.
